src/arrayMethodJoin.js

```diff
@@ -4,8 +4,26 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let newStr = ``;
+
+    for (let i = 0; i < this.length; i++) {
+      let oneWord = this[i];
+
+      if (oneWord === null
+        || oneWord === undefined
+        || oneWord.length === 0) {
+        oneWord = '';
+      }
+
+      if (i === this.length - 1) {
+        newStr += `${oneWord}`;
```

Comment:
Use this[i] instead of variable oneWord and you don't need such type of quotes as you use - ``. Also you can change this quotes into single quotes ' ' in newStr variable in the start.